server/etcdserver/v3_server.go

```diff
@@ -279,6 +285,9 @@ func (s *EtcdServer) LeaseRevoke(ctx context.Context, r *pb.LeaseRevokeRequest)
 }
 
 func (s *EtcdServer) LeaseRenew(ctx context.Context, id lease.LeaseID) (int64, error) {
+	if id < lease.NoLease {
```

Comment:
```suggestion
	if r.ID <= int64(lease.NoLease) {
```

You should not be able to renew NoLease